internal/pkg/workload/workload.go

```diff
@@ -209,13 +209,20 @@ func createDebugContainer(ctx context.Context, client *kubeutil.Client, podName
 			VolumeMounts: []v1.VolumeMount{
 				{
 					ReadOnly:  true,
-					Name:      "spiffe-workload-api",
+					Name:      spiffeSocketVolumeName,
 					MountPath: "/spiffe-workload-api",
 				}},
 		},
 		TargetContainerName: pod.Spec.Containers[0].Name,
 	}
 
+	if spiffeSocketEndpoint != "" {
+		debugContainer.Env = append(debugContainer.Env, v1.EnvVar{
+			Name:  "SPIFFE_ENDPOINT_SOCKET",
```

Comment:
To check: does this work with the existing image? go-spiffe should support this natively.